addons/l10n_sa/models/account_move.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import base64
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    delivery_date = fields.Date(default=fields.Date.context_today, copy=False)
+    show_delivery_date = fields.Boolean(compute='_get_show_delivery_date')
+    qr_code_str = fields.Char(compute='_compute_qr_code_str')
+    confirmation_datetime = fields.Datetime(default=False, readonly=True, copy=False)
+
+    @api.depends('company_id.country_id')
+    def _get_show_delivery_date(self):
+        for move in self:
+            move.show_delivery_date = move.company_id.country_id == self.env.ref('base.sa') and move.move_type in ('out_invoice', 'out_refund', 'out_receipt')
```

Comment:
Compute fields should be named _compute, but directly as attrs in the view (there is a field country_code on the invoice) should be possible as well.

Reply:
The goal of this compute field is to keep it DRY as the same conditions need to be used in 3 different attrs

Reply:
Ok, but try to avoid the ref, but rather country_id.code (or even country_code) then as well.